js/reaction.js

```diff
@@ -101,6 +101,14 @@ function init(reaction) {
   clean();
   // Trigger reaction-level validation.
   validateReaction();
+  // Initialize autosave.
+  setInterval(() => {
+    // Only save if modifications have been made, hence if save button visible.
```

Comment:
I think this would be easier to understand if the thing that enables the "save" button started the timer instead of adding a new timer just to poll the state of the button.

Reply:
Also consider that the user may have clicked the "save" button between the time this timer is started and the time it fires.

Reply:
Implemented your first comment. 

I'm not sure what the best way to work around the problem in your second comment is -- would what I right now be ok? I'm considering between this, and using `clearTimeout`.

Reply:
I would replace the "save" button with a read-only "saving"/"saved" indicator. Then people can't click it at the wrong time.

I would also consider two timers, a short one and a long one. The short one waits before saving in case more are coming. The other forces a save in case a batch takes too long. Definitely hold references to any timers so you can clear them depending on what the user is doing.

Reply:
Could you clarify what you mean by "batch takes too long"? I probably don't know some lingo... (When exactly would we want to save, from the long timer?)
The rest makes sense -- I'll work on those.

Reply:
Also, when an autosave occurs, the "save" button is hidden (until the next change), so users couldn't click it anyways. Would this remove the need for a read-only indicator? Sorry if I'm missing something.

(I started on it, but knowing when a reaction is fully saved is tough, since it's hard to know when all the uploads are finished.)

Reply:
Are we going to be thrashing the binary uploads with this mechanism? That may not be good. Consider the possibility that the operations could overrun each other and cause concurrent uploads.

If we can't know when the operation is done, then we can't prevent overruns, so corruption becomes possible.

Reply:
Ah, that's a good point... Not sure what to do, I'll think about it.

Reply:
We can chat if it might help. Just ping me and I'll probably be free.